Design/SANS/NewISISDesign/Design_Document.md

```diff
@@ -0,0 +1,350 @@
+GUI logic, which essentially renders the ISIS SANS reduction a monolithic block.
+
+A more modern approach is to use Mantid's `WorkflowAlgorithm`s. A sequence of
+algorithms perform the reduction, where each algorithm gets the information
+required for the reduction at the time it executes. The `WorkflowAlgorithm` is
+used as a unit of clearly defined work and **not** as a container for
+state information. This mechanism avoids the deep coupling as each algorithm is
+agnostic about its calling environment and the information flow is into the algorithms
+(except for resulting output workspaces).
+This creates an ideal environment for unit testing and improved documentation.
+
+
+## Reducer 2.0
+
+As per requirement [R.2.4](User_Requirements.md#R.2.4) the reduction of a single
+run should be the limit case of the reduction of a batch file. Currently the
+implementations are slightly different. Unifying the two approaches is beneficial to future maintenance.
+
+### SANS State
+
+The current ISIS SANS system suffers from a state which is distributed and stored
+in many different places. The state for a the SANS reduction system should be
+ localized and complete, i.e. it defines the entire reduction.
+
+A design document for a `SANSState` approach is available
+[here](SANSState/SANSState.md).
+
+The result of this document is that the `SANSState` object contains a
+complete definition of a single reduction (provided the state is valid, which should
+  be checked by validators adequate for the facility/instrument).
+
+### Batch reduction
+
+Starting point of a reduction is a collection of `SANSState` objects which define one
+ reduction per `SANSState`. This batch reduction is handled by a work-flow
+ algorithm `SANSBatchReduction`. It should be able to operate the individual
+ reductions in parallel (does this mean, we need to implement it in `C++`?),
+ which is in line with requirement [R.3.2](#User_Requirements#R.3.2).
+
+The `SANSBatchReduction` performs two tasks:
+
+1. Loads files into the ADS if they are not present yet. These workspaces remain
+in mint condition, i.e. we don't want to move them in order to be able to reuse
+ them. The calibration workspace is being loaded into the ADS as well.
+2. Passes the loaded workspaces and the `SANSState` to a work-flow algorithm,
+which performs a single reduction.
+
+The above ADS feature might be considered unclean, since a work-flow algorithm
+ is accessing the ADS, but this allows us to reuse loaded workspaces which has
+ been expressed as a requirement ([R.3.1](#User_Requirements#R.3.1)). We could think
+ of using the ADS as an optional element.
+
+**Open Question**: What should be done with the output workspaces after each reduction:
+
+  1. Save each workspace to a file after each individual reduction
+  2. Add each workspace to the ADS after each individual reduction
+  3. Collect the output workspaces and save them all at once (probably not)
+  4. Leave this as an option for the user
```

Comment:
I think this should be user-driven, particularly relating to a need for intermediate files. If the answer relates to lack of reliability, then I think you will need to highlight that we expect the new system not to suffer from the same frequency of failures. 

You could solve this with some kind of `persistence` layer such as a `Data Bridge`, which would allow you to implement future strategies for saving an retrieving the information, and adding new ones at a later time.